sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -338,6 +338,15 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     throw new ParseException("INSERT OVERWRITE DIRECTORY is not supported", ctx)
   }
 
+  override def visitDeleteFromTable(
```

Comment:
This needs to create a `DeleteFromStatement`. We use statement logical plans to separate logic in the parser from implementation details.

Reply:
Thanks. Has introduced `DeleteFromStatement` in the latest code as an intermediate object.